### DISCUSSION TITLE
Custom error JSON response structure for Graph QL 22.3 and Spring for Graph QL

### DISCUSSION THREAD

**Discussion by thaparraj** [graphql-java/graphql-java#3716] [Q&A] [answered]
I want a custom response JSON structure:
{
  "errors":[
     {
        "error_code": "",
        "description": ""
     }
  ]
}

How can I do that?

**Comment by bclozel** [comment, marked as answer]
I'm afraid you cannot, as this JSON structure [does not align with the spec](http://spec.graphql.org/draft/#sec-Errors). You can use the "extensions" entry to add custom fields though.

You should see what's possible in the [`GraphQLError` contract](https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/GraphQLError.java) and [how you can turn application exceptions into GraphQL errors in Spring for GraphQL](https://docs.spring.io/spring-graphql/reference/controllers.html#controllers.exception-handler).

**Reply by bbakerman** [reply]
Yup Brian is correct.  According to spec the "extra" attributes need to go into an `extensions` attr

We have errors like this at Atlassian where I work

```
"errrors" : [{
   "message" : "Come back later for more graphql cake",
   "extensions" : {
       "statusCode" : "429",
       "errorSource" : "GATEWAY"
}]
```

It is pretty much what you have above BUT it follows by putting the extra attributes into the `extensions` attribute
